README.md

```diff
@@ -164,27 +161,42 @@ reading the following articles:
 - [Functional Swift for Dealing with Optional Values](http://robots.thoughtbot.com/functional-swift-for-dealing-with-optional-values)
 - [Railway Oriented Programming](http://fsharpforfunandprofit.com/posts/recipe-part2/)
 
+And check out this talk:
+
+- [How I Learned To Stop Worrying And Love The Functor](https://github.com/gfontenot/talks/tree/master/Functors)
+
 ## Usage
 
-The first thing you need to do when you receive JSON data is convert it to an
-instance of the `JSONValue` enum. This is done by passing the `AnyObject`
-value returned from `NSJSONSerialization` to `JSONValue.parse()`:
+The first thing you need to do when you receive JSON data is convert it from
+`NSData` to an `AnyObject` using the build in `NSJSONSerialization` API. Once
+you have the `AnyObject`, you can call the global `decode` function to get back
+the decoded model.
 
 ```swift
 let json: AnyObject? = NSJSONSerialization.JSONObjectWithData(responseData, options: NSJSONReadingOptions(0), error: nil)
 
 if let j: AnyObject = json {
-  let value: JSONValue = JSONValue.parse(j)
-  let user: User? = User.decode(value)
+  let user: User? = decode(j) // ignore error info or
+  let decodedUser: Decoded<User> = decode(j) // preserver error info
 }
 ```
```

Comment:
While we're at it, I think we should modify this code snippet so that it actually uses an error pointer and checks the return value, instead of just punting and mentioning in the following paragraph "this it what you should have done instead". I understand it's a simple example, but just a couple additional lines of code will make it more correct without sacrificing simplicity too much, IMHO.